nav2_simple_commander/test/test_footprint_collision_checker.py

```diff
@@ -0,0 +1,139 @@
+# Copyright 2022 Afif Swaidan
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+from nav2_simple_commander.footprint_collision_checker import FootprintCollisionChecker
+from nav2_simple_commander.costmap_2d import PyCostmap2D
+from nav_msgs.msg import OccupancyGrid
+from geometry_msgs.msg import Polygon
+from geometry_msgs.msg import Point32
+
+LETHAL_OBSTACLE = 254
+
+
+class TestFootprintCollisionChecker(unittest.TestCase):
```

Comment:
Last thing : the unit tests should have some that have non-zero costs in them to make sure things are working properly. If you make a 100x100 map with a 10x10 block of cost in the middle (or something) and then run these same tests to make sure we're picking things up correctly.

Then this is good to merge! Thanks!